src/main/java/com/teentech/hotels/controller/BarController.java

```diff
@@ -0,0 +1,86 @@
+package com.teentech.hotels.controller;
+
+import com.teentech.hotels.dto.BarDto;
+import com.teentech.hotels.model.Bar;
+import com.teentech.hotels.model.HotelRoomsPK;
+import com.teentech.hotels.model.Reservations;
+import com.teentech.hotels.service.BarService;
+import com.teentech.hotels.service.MailService;
+import com.teentech.hotels.service.ReservationService;
+import com.teentech.hotels.util.BarConverter;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.mail.BodyPart;
+import javax.mail.Multipart;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMultipart;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/bar")
+@Log4j2
+public class BarController {
+
+    @Autowired
+    private BarService barService;
+
+    @Autowired
+    private ReservationService reservationService;
+
+    @GetMapping
+    public ResponseEntity<BarDto> getBarById(@RequestParam long hotelId, @RequestParam long roomNumber) {
+        try {
+            HotelRoomsPK hotelRoomsPK = HotelRoomsPK.builder().hotelId(hotelId).roomNumber(roomNumber).build();
+            Bar bar = barService.findBarById(hotelRoomsPK);
+            BarDto barDto = BarConverter.convertFromEntityToDto(bar);
+            return new ResponseEntity<>(barDto, HttpStatus.OK);
+        } catch (Exception e) {
+            log.error("Error while getting bar info", e);
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @PutMapping
+    public ResponseEntity<Boolean> updateBar(@RequestBody BarDto barDto) {
+        try {
+            Bar bar = BarConverter.convertFromDtoToEntity(barDto);
+            barService.update(bar);
+
+            String htmlText = "<h1 style=\"text-align:center;  font-family:'FranklinGothicMedium','ArialNarrow',Arial,sans-serif;" +
+                    "font-weight:100;font-size:2vw;color:#0e77de;margin-top:3vw;margin-bottom:3vw;\"> &#127867; This is your Drinking Status &#127867;</h1>" +
+                    "<p style=\"text-align:center;font-family:'GillSans','GillSansMT',Calibri,'TrebuchetMS',sans-serif;font-size:1vw;\">" +
+                    "You have <b>" + bar.getNoOfDrinks() + " more drinks</b> to savor. &#128513;</p>";
+
+            BodyPart messageBodyPart = new MimeBodyPart();
```

Comment:
trebuie facuta o clasa EmailService in care sa trimiti un obiect cu contentu, subject, adresa destinatar si sa fie utilizata peste tot